### PR TITLE
MacOS install Xcode 15 fails

### PR DESCRIPTION
**Pull request by Ig-dolci**
Some MacOS users are encountering fails to install Firedrake with Xcode15. Apparently, Xcode 15 beta has sorted it. Should be this informed in an install note?

### CODE REVIEW

**Review by JDBetteridge — CHANGES_REQUESTED** [review]
Maybe this could be added to the flowchart? https://www.firedrakeproject.org/download.html#debugging-install-problems. 

At the very least it should be moved down from the top section to debugging install problems section. We should bring this up at the meeting to try and find out how widespread this issue really is...

**Comment by Ig-dolci** [comment]
> Maybe this could be added to the flowchart? https://www.firedrakeproject.org/download.html#debugging-install-problems.
> 
> At the very least it should be moved down from the top section to debugging install problems section. We should bring this up at the meeting to try and find out how widespread this issue really is...

Thank you, Jack. I will add this issue in the next meeting.

File: docs/source/download.rst [hunk 1]
```diff
@@ -212,6 +212,14 @@ Likewise, if it's ``firedrake-update`` that fails, please include the
 file ``firedrake-update.log``. You can find this in the Firedrake
 virtual environment.
 
+.. note::
+
+  Some MacOS users have experienced difficulties on installing firedrake on Xcode 15,
+  and updating from Xcode 15 to the beta version has been a solution. If you are
+  encountering this issue, we recommend attempting this update as a potential solution.
+  Keep us informed if you face any challenges or if this approach does not resolve the
+  install issue.
```
Comment (dham) [inline docs/source/download.rst:221]: please tell them how.

Comment (Ig-dolci) [reply docs/source/download.rst]: Done.

**Comment by JDBetteridge** [comment]
Can we have a bit of a tidy up of various MacOS related fixes on the wiki, rather than just linking to another random page?

https://github.com/firedrakeproject/firedrake/wiki/Fixing-venv-on-macOS-after-homebrew-patches-Python
https://github.com/firedrakeproject/firedrake/wiki/Installing-pyenv
https://github.com/firedrakeproject/firedrake/wiki/macOS-version-support
and now
https://github.com/firedrakeproject/firedrake/wiki/Installation-Issues-with-Firedrake-on-Xcode-15-for-MacOS-Users

All contain various miscellaneous fixes related to MacOS. We should determine which advice is still relevant, consolidate this into one page and then add a link to the wiki sidebar and then link to that document from the Firedrake website.

File: docs/source/download.rst [hunk 2]
```diff
@@ -212,6 +212,13 @@ Likewise, if it's ``firedrake-update`` that fails, please include the
 file ``firedrake-update.log``. You can find this in the Firedrake
 virtual environment.
 
+.. note::
+  MacOS users experiencing firedrake installation issues on Xcode 15 
+  can potentially resolve them by updating to the beta version of Xcode 15. 
+  The instructions for the Xcode 15 update are available on this link `link
```
Comment (connorjward) [inline docs/source/download.rst:218]:
```suggestion
  The instructions for the Xcode 15 update are available on this `link 
```

File: docs/source/download.rst [hunk 3]
```diff
@@ -212,6 +212,13 @@ Likewise, if it's ``firedrake-update`` that fails, please include the
 file ``firedrake-update.log``. You can find this in the Firedrake
 virtual environment.
 
+.. note::
+  MacOS users experiencing firedrake installation issues on Xcode 15 
+  can potentially resolve them by updating to the beta version of Xcode 15. 
+  The instructions for the Xcode 15 update are available on this link `link 
+  <https://github.com/firedrakeproject/firedrake/wiki/Installation-Issues-with-Firedrake-on-Xcode-15-for-MacOS-Users>`_
```
Comment (connorjward) [inline docs/source/download.rst:219]:
```suggestion
  <https://github.com/firedrakeproject/firedrake/wiki/Installation-Issues-with-Firedrake-on-Xcode-15-for-MacOS-Users>`_.
```

File: docs/source/download.rst (hunk 2 above)
Comment (connorjward) [reply docs/source/download.rst:218]: says "link" twice.

**Review by connorjward — CHANGES_REQUESTED** [review]

**Comment by Ig-dolci** [comment]
> Can we have a bit of a tidy up of various MacOS related fixes on the wiki, rather than just linking to another random page?
> 
> https://github.com/firedrakeproject/firedrake/wiki/Fixing-venv-on-macOS-after-homebrew-patches-Python https://github.com/firedrakeproject/firedrake/wiki/Installing-pyenv https://github.com/firedrakeproject/firedrake/wiki/macOS-version-support and now https://github.com/firedrakeproject/firedrake/wiki/Installation-Issues-with-Firedrake-on-Xcode-15-for-MacOS-Users
> 
> All contain various miscellaneous fixes related to MacOS. We should determine which advice is still relevant, consolidate this into one page and then add a link to the wiki sidebar and then link to that document from the Firedrake website.

Yes. I agree that we should organise these instructions.
I think that the instruction https://github.com/firedrakeproject/firedrake/wiki/Fixing-venv-on-macOS-after-homebrew-patches-Python can be removed. I just created a [wiki page](https://github.com/firedrakeproject/firedrake/wiki) `MacOS firedrake install`.

**Comment by connorjward** [comment]
I think that rather than having an "everything MacOS" page we should have an FAQ style wiki document on "Fixing install issues" that we can link to from the website. We should split it into "Current" and "Archived" sections or similar.

**Comment by connorjward** [comment]
Also https://github.com/firedrakeproject/firedrake/wiki/macOS-version-support should be kept in the policies and procedures section of the wiki.

**Comment by JDBetteridge** [comment]
Sure, although mention of the mystery `MACOSX_DEPLOYMENT_TARGET` environment variable should be mentioned on the MacOS help page.

**Pull request closed without merging**